sw/device/silicon_creator/lib/log_unittest.cc

```diff
@@ -0,0 +1,85 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/silicon_creator/lib/log.h"
+
+#include "gtest/gtest.h"
+#include "sw/device/silicon_creator/lib/drivers/uart.h"
+#include "sw/device/silicon_creator/lib/error.h"
+#include "sw/device/silicon_creator/testing/mask_rom_test.h"
+
+namespace log_unittest {
+
+namespace internal {
+// Create a mock for shutdown functions.
+// TODO: move mock UART into its own header file.
+class MockUart : public ::mask_rom_test::GlobalMock<MockUart> {
+ public:
+  MOCK_METHOD(void, uart_putchar, (char));
+};
+}  // namespace internal
+
+using MockUart = testing::StrictMock<internal::MockUart>;
+extern "C" {
+void uart_putchar(uint8_t c) { MockUart::Instance().uart_putchar(c); }
+}  // extern "C"
+
+class LogTest : public mask_rom_test::MaskRomTest {
+ protected:
+  void ExpectPrint(std::string str) {
```

Comment:
Util we get absl::string_view, how about setting this to `const std::string &str`?